src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -479,15 +479,18 @@ def get_platform_info(config_db=None):
     if hw_info_dict:
         return hw_info_dict
 
-    from .multi_asic import get_num_asics
+    from .multi_asic import get_asic_presence_list
```

Comment:
can you explain the reason for this change? why get_num_asics does not work, but get_asic_presence_list?

what is the scenario that was broken?

Reply:
@lguohan 
In Chassis for supervisor running asic count can be <= max asic possible.
`get_num_asics()` return max asic count that can be present based on static file eg: https://github.com/sonic-net/sonic-buildimage/blob/fe24c2699681df2e3ef86199718b5e22651e6eac/device/nokia/x86_64-nokia_ixr7250e_sup-r0/asic.conf#L1. However in supervisor max_asic represnt number of asic where all Fabric cards are populated. However most of the time we have less than that inserted. `get_asic_presence_list()` returns how many asics are detected on Supervisor and for LC/Multi-asic system it still return max asic count